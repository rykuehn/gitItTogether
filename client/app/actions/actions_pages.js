export const CREATE_ISSUES_LIST = 'CREATE_ISSUES_LIST';
export const GET_CURRENT_PAGE = 'GET_CURRENT_PAGE';
export const DISPLAY_ISSUE = 'DISPLAY_ISSUE';

export function issues(data) {
  const issuesPerPage = 25;
  const numOfPages = Math.ceil(data.length / issuesPerPage);
  localStorage.setItem('issues', JSON.stringify(data));
  return {
    type: CREATE_ISSUES_LIST,
    payload: {
      data,
      numOfPages,
      issuesPerPage,
    },
  };
}

export function currentPage(page) {
  return {
    type: GET_CURRENT_PAGE,
    page,
  };
}

export function displayIssue(page, positionOnCurrentPage) {
  return {
    type: DISPLAY_ISSUE,
    payload: {
      page,
      positionOnCurrentPage,
    },
  };
}
