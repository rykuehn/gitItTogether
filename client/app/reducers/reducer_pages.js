import { combineReducers } from 'redux'
import _ from 'underscore';
import { REQUEST_PAGE, RECEIVE_PAGE } from '../actions/actions_pages';

export function pages(pages ={}, action) {
  const pageCopy = _.extend({}, pages);
  switch(action.type) {
    case REQUEST_PAGE:
      return 
        pageCopy[action.payload.page] = {
          id: [],
          fetching: true
        }
    case RECEIVE_PAGE:
      return 
        pageCopy[action.payoad.page] = {
          ids: action.payload.results.filter(item => item.id),
          fetching: false
        
      }
    default: 
      return pages;
  }
}

export function currentPage(currentPage = 1, action = {}){
  return action.type == REQUEST_PAGE ? action.payload.page : currentPage
}

// const todos = (todos = {}, action = {}) {
//   switch (action.type) {
//     case 'RECEIVE_ALL_TODOS':
//     case 'RECEIVE_TODO_PAGE':
//       let _todos = {}
//       for (let todo of action.payload.todos) {
//         _todos = {
//           ..._todos,
//           [todo.id]: todo
//         }
//       }
//       return {
//         ...todos,
//         ..._todos
//       }
//     default:
//       return todos
//   }
// }

