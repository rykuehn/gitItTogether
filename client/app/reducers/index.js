import { combineReducers } from 'redux';
import { routerReducer } from 'react-router-redux';
import { pagesReducer, currentPageReducer, totalPagesReducer, currentIssueDisplayReducer } from './reducer_pages';

const rootReducer = combineReducers({
  pagesReducer,
  currentPageReducer,
  totalPagesReducer,
  currentIssueDisplayReducer,
  routing: routerReducer,
});

export default rootReducer;
