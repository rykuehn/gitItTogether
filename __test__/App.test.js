import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import ConnectedApp, { App } from '../client/app/components/app';
import { issues, CREATE_ISSUES_LIST } from '../client/app/actions/actions_pages';
import { totalPagesReducer } from '../client/app/reducers/reducer_pages';
import mockData from '../client/app/mock';

describe('Heading renders correctly', () => {
  const rendered = renderer.create(<App />).toJSON();
  expect(rendered).toMatchSnapshot();
});

describe('app shallow render', () => {
  const initialState = { pageCount: 0 };
  const mockStore = configureStore();
  let component;

  beforeEach(() => {
    let store = mockStore(initialState);
    component = mount(<Provider store={store}><ConnectedApp /></Provider>);
  });

  it('render smart component', () => {
     expect(component.find(ConnectedApp).length).toEqual(1);
  });
});

describe('Reducers', () => {
  it('Total Pages Reducer', () => {
    let initialState = { pageCount: 0 };
    initialState = totalPagesReducer(initialState, {
      type: CREATE_ISSUES_LIST,
      payload: {
        data: mockData,
        numOfPages: 2,
        issuesPerPage: 25,
      } });

    expect(initialState).toEqual(2);
  });
});
