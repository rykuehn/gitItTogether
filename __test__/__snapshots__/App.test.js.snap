// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`undefined 1`] = `
<div
  style={
    Object {
      "alignContent": "space-between",
      "alignItems": "center",
      "backgroundColor": "#F7EDE2",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "center",
      "margin": "0px",
      "width": "100%",
    }
  }
>
  
  <div />
</div>
`;
